js/util.ts

```diff
@@ -151,3 +151,15 @@ export function requiredArguments(
     throw new TypeError(errMsg);
   }
 }
+
+// Returns values from a WeakMap to emulate private properties in JavaScript
+export function getPrivateValue<
+  K extends object,
+  V extends object,
+  W extends keyof V
+>(instance: K, weakMap: WeakMap<K, V>, key: W, defaultValue: V[W]): V[W] {
```

Comment:
This is my only comment, the `defaultValue` is not being used in the helper function and it should be removed.